### DISCUSSION TITLE
Force field to calculate value before submission

### DISCUSSION THREAD

**Discussion by equ8r** [Budibase/budibase#5799] [Help] [answered]
I'm new to the platform... very nice.

I've got a series of form fields that collect input from users, and then a totals field that grabs the provided form data and calculates the total. The issue is it passes through blank if the user clicks the 'save' button with having focused on the totals field. How can I force the field to calculate as part of the save/submit process?

I have tried calculating the value in the field default value setting (which works if the user focuses on the field prior to saving) as well as defining an on-click action to update the field's value prior to validation. It passes through as blank, but if I click in to edit the created record it immediately populates with the correct calculated value. 

A kick in the right direction would be much appreciated. :)

**Comment by melohagan** [comment, marked as answer]
Hey @equ8r 

You don't necessarily need to grab the value from the totals field. Why not just calculate the total when you click 'Save'?

In your 'Save Row' action you can click 'Add Column' and then have your sum function in there as the binding for your table's total field.

**Reply by equ8r** [reply]
That worked as I intended. Thanks for the assist!

I'm still getting up to speed with where everything is... Back with more rudimentary questions in short order, I suspect. :)

**Reply by melohagan** [reply]
You're very welcome, thanks for giving Budibase a whirl!
